test/com/amazon/ion/impl/bin/WriteBufferTest.java

```diff
@@ -281,6 +263,20 @@ public void testInt56Negative()
         assertBuffer(bytes);
     }
 
+    @Test
+    public void testEndOfBufferReachedInvoked() throws UnsupportedEncodingException {
+        buf.writeBytes("taco".getBytes("UTF-8"));
+        buf.writeBytes("_burrito".getBytes("UTF-8"));
+        assertTrue(endOfBufferReached.get());
+    }
+
+    @Test
+    public void testEndOfBufferReachedNotInvoked() throws UnsupportedEncodingException {
+        buf.writeBytes("taco".getBytes("UTF-8"));
+        buf.writeBytes("burrito".getBytes("UTF-8"));
+        assertFalse(endOfBufferReached.get());
+    }
```

Comment:
Please add some comments to these tests. It's not immediately clear why adding the underscore changes the behavior. Consider noting the block size, and add assertions before and after the boundary to verify the transition happens exactly when expected.

Reply:
> Please add some comments to these tests. It's not immediately clear why adding the underscore changes the behavior. Consider noting the block size, and add assertions before and after the boundary to verify the transition happens exactly when expected.

Sure, I will add the comment in the next commit. If we are adding assert before and after the boundary, we might not need two unit tests.